tools/nimgrep.nim

```diff
@@ -115,20 +157,23 @@ type
                     lineBeg: int, lineEnd: int, match: string]
   outputKind = enum
     OpenError, Rejected, JustCount,
-    BlockFirstMatch, BlockNextMatch, BlockEnd, FileContents
+    BlockFirstMatch, BlockNextMatch, BlockEnd, FileContents, FileName
   Output = object
     case kind: outputKind
-    of OpenError: msg: string
-    of Rejected: reason: string
-    of JustCount: matches: int
-    of BlockFirstMatch, BlockNextMatch:
+    of OpenError: msg: string           # file/directory not found
+    of Rejected: reason: string         # when the file contents do not pass
+    of JustCount: matches: int          # the only output for option --count
+    of BlockFirstMatch, BlockNextMatch: # the normal case: match itself
       pre: string
       match: MatchInfo
-    of BlockEnd:
+    of BlockEnd:                        # block ending right after prev. match
       blockEnding: string
-      firstLine: int       # = last lineNo of last match
-    of FileContents:
+      firstLine: int
```

Comment:
0 or 1 based indexing?

Reply:
AFAIK line numbers are always numbered from 1 in all text editors, so I assumed that it goes without saying